### ISSUE TITLE
Support for LIKE operator

### ISSUE THREAD

**Issue by dasim** [nextras/orm#250]
Hi, great library, using it for a while successfully, so thanks for your work!

I wondered, will there be (or am I missing already) support for easy conditions with LIKE %% operator? My need is a simple and easy search by `name LIKE %something%`. I currently use specific method in a mapper that’s made available by annotation in corresponding repository. It works fine.

Here is an example of such simple mapper:

```php
<?php

namespace App\Model;

class UserMapper extends \Nextras\Orm\Mapper\Mapper
{
	protected $tableName = 'user';

	public function filterSearch($params)
	{
		$qb = $this->builder();
		if (isset($params['search_name']) && $params['search_name']) {
			if (!empty($terms = $this->splitString($params['search_name']))) {
				foreach ($terms as $t) {
					$qb->andWhere($this->tableName . '.name LIKE %_like_', $t);
				}
			}
		}
		return $this->toCollection($qb);
	}


	private function splitString($string)
	{
		$string = $string ? : '';
		return array_filter(array_map(function ($t) {
			return trim($t);
		}, explode(' ', $string)));
	}
}
```

I’m currently testing v3.0.0 beta and it works just fine. But when I get to more complex conditions using `ICollection::OR` operator, my mapper gets more complex as well by trying to find the `search_name` parameter and also I’m not really sure if it will work well with `->andWhere` part in my `filterSearch` method.

It would be great if it would be possible to use something like `$this->orm->users->findBy(['name LIKE' => '%something%'])` or similar instead. Do you think it’s a feasible to ask for that?

**Comment by hrach** [comment]
Support for this was implemented through Custom functions, which are not documented yet (a new feature in 3.0), take a look here: https://github.com/nextras/orm/issues/127#issuecomment-324424227

**Comment by dasim** [comment]
Cool, thanks a lot. I think I was on the right path with Custom functions while exploring v3 but lack of documentation stopped. Great to know it’s coming!

**Comment by dasim** [comment]
Closing, will be finished by #251.

**Comment by hrach** [comment]
LIKE support was just merged into 4.0. See PR #416.
